packages/docs/docsite/src/components/splash/splash.tsx

```diff
@@ -0,0 +1,357 @@
+							},
+							'PieceTogether+=' + Math.random() * 0.3
+						)
+					}
+				}
+
+				tl.to(
+					'.background',
+					0.3,
+					{
+						opacity: 0,
+						easing: Elastic,
+						delay: 0.2,
+					},
+					'Scale' + 0
+				)
+
+				setTimeout(() => {
+					const svgElem = document.getElementById('svg-logo2')!
+					const cdims2 = svgElem.getBoundingClientRect()
+					const offset = document
+						.getElementById('chart-parts-letter-group')!
+						.getBoundingClientRect()
+					const callbackFunc = () => {
+						setAnimationOut(true)
+					}
+					const timeLineOut = new TimelineMax({
+						onComplete: callbackFunc,
+						delay: 0.1,
+					})
+					timeLineOut.addLabel('Scale', 0.4)
+					timeLineOut.addLabel('Links', 0.4)
+					timeLineOut.to(
+						svgElem,
+						0.4,
+						{
+							x: -(
+								cdims2.left +
+								(offset.left - cdims2.left) +
+								(cdims2.left + (offset.left - cdims2.left)) * 1.8
+							),
+							y: -(
+								cdims2.top +
+								(offset.top - cdims2.top) +
+								offset.height / 2 +
+								10
+							),
+							scale: 0.2,
+						},
+						'Scale'
+					)
+					timeLineOut.to(
+						title,
+						0.4,
+						{
+							opacity: 0,
+							delay: 0.35,
+						},
+						'Scale'
+					)
+					timeLineOut.to(
+						'.header',
+						0.1,
+						{
+							opacity: 1,
+							delay: 0.35,
+						},
+						'Scale'
+					)
+					// stagger in fade links right to left
+					const links = ['links', 'source', 'docs', 'blog']
+					links.forEach((item: string, index: number) => {
+						timeLineOut.to(
+							`#${item}`,
+							0.2,
+							{
+								opacity: 1,
+								delay: 0.4 + index * 0.1,
+							},
+							'Links+=' + 0
+						)
+					})
+
+					timeLineOut.play()
+				}, 1000)
+			}
+		}
+		animate()
+	}
+
+	// get all chart part elements
+	const getChartPartsChars = () => {
+		const chars = ['c', 'h', 'a', 'r', 't', 'p', 'a', 'r', 't', 's']
+		return chars.reduce((acc: HTMLElement[], letter: string, index: number) => {
+			const elem = document.getElementById(`char-${letter}${index + 1}`)!
+			acc.push(elem)
+			return acc
+		}, [])
+	}
+
+	// highlight div on mouseover
+	const mouseEnter = useCallback(
+		(event: React.MouseEvent<any>, ref: any) => {
+			if (ref && ref.current && setAnimationOut) {
+				anime({
+					targets: ref.current,
+					easing: 'easeOutExpo',
+					opacity: [0.8, 1],
+					duration: 500,
+				})
+			}
+		},
+		[setAnimationOut]
+	)
+
+	// return div to prev opacity
+	const mouseLeave = useCallback(
+		(event: React.MouseEvent<any>, ref: any) => {
+			if (ref && ref.current && setAnimationOut) {
+				anime({
+					targets: ref.current,
+					easing: 'easeOutExpo',
+					opacity: [1, 0.8],
+					duration: 500,
+				})
+			}
+		},
+		[setAnimationOut]
+	)
+
+	const boxRowHeight = useMemo(() => {
+		return `${height - (headerHeight + FOOTER_HEIGHT + 10)}px`
+	}, [height])
+
+	const variableOpacity = useMemo(() => {
+		return animatingOut ? 1 : 0
```

Comment:
this statement is terse enough to not need a `return` statement or braces: `useMemo(() => animatingOut ? 1 : 0, [animatingOut)`